src/main/java/spoon/support/compiler/jdt/ReferenceBuilder.java

```diff
@@ -1188,7 +1188,11 @@ private CtTypeReference<?> getTypeReferenceFromProblemReferenceBinding(ProblemRe
 
 		CtTypeReference<?> ref = this.jdtTreeBuilder.getFactory().Core().createTypeReference();
 		ref.setSimpleName(stripPackageName(readableName));
-		final CtReference declaring = this.getDeclaringReferenceFromImports(binding.sourceName());
+		CtReference declaring = this.getDeclaringReferenceFromImports(binding.sourceName());
+		String packageName = getPackageName(readableName);
```

Comment:
Maybe rename this to something indicating it is optional (or just make `getPackageName` return an `Optional<String>` in the first place)?